SandWorm/SandWormComponent.cs

```diff
@@ -138,190 +139,185 @@ protected override void SolveInstance(IGH_DataAccess DA)
+            }
+
+            // Setup variables for per-pixel loop
+            pointCloud = new Point3d[trimmedWidth * trimmedHeight];
+            int arrayIndex = 0;
+            for (int rows = 0; rows < trimmedHeight; rows++)
+            {
+                for (int columns = 0; columns < trimmedWidth; columns++)
+                {
+                    depthPoint = averagedDepthFrameData[arrayIndex];
+                    tempPoint.X = columns * -unitsMultiplier * depthPixelSize.x;
+                    tempPoint.Y = rows * -unitsMultiplier * depthPixelSize.y;
+                    tempPoint.Z = (depthPoint - sensorElevation) * -unitsMultiplier;
+                    pointCloud[arrayIndex] = tempPoint; // Add new point to point cloud itself
+                    arrayIndex++;
                 }
+            }
+            Core.LogTiming(ref output, timer, "Point cloud generation"); // Debug Info
+            
+            // First type of analysis that acts on the pixel array and produces vertex colors
+            vertexColors = new Color[pointCloud.Length];
+            switch (Analysis.AnalysisManager.GetEnabledMeshColoring())
+            {
+                case Analytics.None analysis:
+                    break;
+                case Analytics.Elevation analysis:
+                    analysis.GetColorCloudForAnalysis(ref vertexColors, averagedDepthFrameData);
+                    break;
+                case Analytics.Slope analysis:
+                    analysis.GetColorCloudForAnalysis(ref vertexColors, averagedDepthFrameData,
+                        trimmedWidth, trimmedHeight, depthPixelSize.x, depthPixelSize.y);
+                    break;
+                case Analytics.Aspect analysis:
+                    // TODO: implementation
+                    break;
+                default:
+                    break;
+            }
+            Core.LogTiming(ref output, timer, "Point cloud analysis"); // Debug Info
+            
+            // Keep only the desired amount of frames in the buffer
+            while (renderBuffer.Count >= averageFrames)
+            {
+                renderBuffer.RemoveFirst();
+            }
 
-                DA.SetDataList(0, outputMesh);
-                DA.SetDataList(1, outputGeometry);
-                DA.SetDataList(2, output); //debugging
+            quadMesh = Core.CreateQuadMesh(quadMesh, pointCloud, vertexColors, trimmedWidth, trimmedHeight);
+            outputMesh.Add(quadMesh);
+            Core.LogTiming(ref output, timer, "Meshing"); // Debug Info
+
+            // Second type of analysis that acts on the mesh and produces new geometry
+            outputGeometry = new List<Rhino.Geometry.GeometryBase>();
+            foreach (var enabledAnalysis in Analysis.AnalysisManager.GetEnabledMeshAnalytics())
+            {
+                switch (enabledAnalysis)
+                {
+                    case Analytics.Contours analysis:
```

Comment:
contourInterval has to be a double otherwise it doesn't work if units are other than millimeters.